jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostEstimator.scala

```diff
@@ -107,18 +107,33 @@ class XGBoostEstimator private[spark](
     }
   }
 
+  private def ensureColumns(trainingSet: Dataset[_]): Dataset[_] = {
```

Comment:
Yes, but to me this seems to be an OK behaviour, since base margin is not required for learning.